pyinfra/api/operation.py

```diff
@@ -306,50 +331,29 @@ def _solve_operation_consistency(names, state, host):
     return op_order, op_hash
 
 
-def _generate_operation_name(func, host, kwargs, global_kwargs):
-    # Generate an operation name if needed (Module/Operation format)
-    name = global_kwargs.get("name")
-    add_args = False
-    if name:
-        names = {name}
-    else:
-        add_args = True
-
-        if func.__module__:
-            module_bits = func.__module__.split(".")
-            module_name = module_bits[-1]
-            name = "{0}/{1}".format(module_name.title(), func.__name__.title())
-        else:
-            name = func.__name__
-
-        names = {name}
-
-    if host.current_deploy_name:
-        names = {"{0} | {1}".format(host.current_deploy_name, name) for name in names}
-
-    return names, add_args
+def _ensure_shared_op_meta(state, op_hash, op_order, global_kwargs, names):
```

Comment:
Let's add a comment indicating this function modifies state.

Reply:
https://github.com/Fizzadar/pyinfra/pull/883/commits/f30aa877f7cdb5349d0f41f3459886c3255532af